File: app/src/screens/PremiumCodeScreen.js
// PREMIUM KODU EKRANI — kullanıcı, kendisine verilen kodu buraya yazar.
// ---------------------------------------------------------------------------
// AKIŞ: operatör yönetim panelinden kod üretir → kullanıcıya iletir →
// kullanıcı burada kullanır → hak sunucuda yazılır.
//
// DÜRÜSTLÜK KURALLARI
//  • Durum SUNUCUDAN okunur. Uygulama "premium" bilgisini kendi belleğinde
//    tutup göstermez; kapatıp açınca farklı bir şey söylemesin.
//  • Premium sistemi sunucuda henüz kurulmadıysa (tablolar yok) ekran bunu
//    açıkça yazar — "premium değilsin" deyip sebebi gizlemez.
//  • Hata mesajı sunucudan ne geldiyse odur; uydurulmuş "başarılı" yok.
import React, { useCallback, useEffect, useState } from 'react';
import {
  View, Text, TextInput, TouchableOpacity, ScrollView, StyleSheet, ActivityIndicator,
} from 'react-native';

import { api } from '../api';
import { colors, spacing, radius } from '../theme';

/** Kullanıcı kodu tireli/küçük harfli yazabilir; sunucuya kanonik biçim gider. */
function kodTemizle(ham) {
  return String(ham || '').toUpperCase().replace(/[^A-Z0-9]/g, '');
}

export default function PremiumCodeScreen() {
  const [durum, setDurum] = useState(null);
  const [yukleniyor, setYukleniyor] = useState(true);
  const [kod, setKod] = useState('');
  const [gonderiliyor, setGonderiliyor] = useState(false);
  const [mesaj, setMesaj] = useState(null);      // { tur: 'ok'|'hata', metin }

  const durumOku = useCallback(async () => {
    setYukleniyor(true);
    try {
      setDurum(await api.premiumDurum());
    } catch (e) {
      setDurum(null);
      setMesaj({ tur: 'hata', metin: e?.message || 'Durum okunamadı.' });
    } finally {
      setYukleniyor(false);
    }
  }, []);

  useEffect(() => { durumOku(); }, [durumOku]);

  const kullan = useCallback(async () => {
    const temiz = kodTemizle(kod);
    if (temiz.length < 4) { setMesaj({ tur: 'hata', metin: 'Kodu eksiksiz yaz.' }); return; }
    setGonderiliyor(true);
    setMesaj(null);
    try {
      const r = await api.premiumKodKullan(temiz);
      setDurum(r);
      setKod('');
      setMesaj({ tur: 'ok', metin: 'Kod kullanıldı — premium erişimin açıldı.' });
    } catch (e) {
      setMesaj({ tur: 'hata', metin: e?.message || 'Kod kullanılamadı.' });
    } finally {
      setGonderiliyor(false);
    }
  }, [kod]);

  const premium = durum?.premium === true;

  return (
    <ScrollView style={s.kok} contentContainerStyle={{ padding: spacing.md, paddingBottom: 40 }}>
      <View style={[s.kart, premium && s.kartPremium]}>
        <Text style={s.baslik}>{premium ? '⭐ Premium erişimin açık' : 'Premium erişimin yok'}</Text>
        {yukleniyor ? (
          <ActivityIndicator color={colors.primary} style={{ marginTop: 8 }} />
        ) : (
          <Text style={s.aciklama}>
            {durum?.kurulmadi
              ? 'Premium sistemi bu sunucuda henüz kurulmadı.'
              : premium
                ? (durum.suresiz
                  ? 'Süresiz erişimin var.'
                  : `Bitiş: ${durum.bitis ? new Date(durum.bitis).toLocaleDateString('tr-TR') : 'bilinmiyor'}`)
                : 'Elinde bir kod varsa aşağıya yazarak erişimini açabilirsin.'}
          </Text>
        )}
      </View>

      <View style={s.kart}>
        <Text style={s.baslik}>Kodum var</Text>
        <TextInput
          style={s.giris}
          value={kod}
          onChangeText={setKod}
          placeholder="ÖRNEK: A7K2M9P4XR"
          placeholderTextColor={colors.textMuted}
          autoCapitalize="characters"
          autoCorrect={false}
          maxLength={24}
          accessibilityLabel="Premium kodu"
        />
        <TouchableOpacity
          style={[s.dugme, gonderiliyor && { opacity: 0.6 }]}
          onPress={kullan}
          disabled={gonderiliyor}
          activeOpacity={0.85}
        >
          <Text style={s.dugmeTxt}>{gonderiliyor ? 'Kontrol ediliyor…' : 'Kodu kullan'}</Text>
        </TouchableOpacity>

        {mesaj ? (
          <Text style={[s.mesaj, mesaj.tur === 'ok' ? s.mesajOk : s.mesajHata]}>{mesaj.metin}</Text>
        ) : null}

        <Text style={s.not}>
          Kodlar büyük/küçük harf ve tire farkına takılmaz; “a7k2-m9p4” ile “A7K2M9P4” aynıdır.
          Bir kodu yalnız bir kez kullanabilirsin.
        </Text>
      </View>
    </ScrollView>
  );
}

const s = StyleSheet.create({
  kok: { flex: 1, backgroundColor: colors.bg },
  kart: {
    backgroundColor: colors.card, borderRadius: radius.md, borderWidth: 1,
    borderColor: colors.border, padding: spacing.md, marginBottom: spacing.md,
  },
  kartPremium: { borderColor: colors.warning, borderWidth: 2 },
  baslik: { color: colors.text, fontSize: 15, fontWeight: '900' },
  aciklama: { color: colors.textSoft, fontSize: 13, marginTop: 6, lineHeight: 18 },
  giris: {
    marginTop: 10, borderWidth: 1, borderColor: colors.border, borderRadius: radius.sm,
    paddingHorizontal: 12, paddingVertical: 11, color: colors.text,
    fontSize: 16, fontWeight: '900', letterSpacing: 2, backgroundColor: colors.surfaceSoft,
  },
  dugme: {
    marginTop: 10, backgroundColor: colors.primary, borderRadius: radius.sm,
    paddingVertical: 12, alignItems: 'center',
  },
  dugmeTxt: { color: '#fff', fontSize: 14, fontWeight: '900' },
  mesaj: { marginTop: 10, fontSize: 13, fontWeight: '800' },
  mesajOk: { color: colors.success },
  mesajHata: { color: colors.accent },
  not: { color: colors.textMuted, fontSize: 11.5, lineHeight: 16, marginTop: 10 },
});

File: flutter/lib/features/profile/premium_code_screen.dart
// KAYNAK: app/src/screens/PremiumCodeScreen.js — BİREBİR çeviri.
//
// PREMIUM KODU EKRANI — kullanıcı, kendisine verilen kodu buraya yazar.
// ---------------------------------------------------------------------------
// AKIŞ: operatör yönetim panelinden kod üretir → kullanıcıya iletir →
// kullanıcı burada kullanır → hak sunucuda yazılır.
//
// DÜRÜSTLÜK KURALLARI
//  • Durum SUNUCUDAN okunur. Uygulama "premium" bilgisini kendi belleğinde
//    tutup göstermez; kapatıp açınca farklı bir şey söylemesin.
//  • Premium sistemi sunucuda henüz kurulmadıysa (tablolar yok) ekran bunu
//    açıkça yazar — "premium değilsin" deyip sebebi gizlemez.
//  • Hata mesajı sunucudan ne geldiyse odur; uydurulmuş "başarılı" yok.

import 'package:flutter/material.dart';

import '../../core/network/api_client.dart';
import '../../core/theme/tokens.dart';

/// Kullanıcı kodu tireli/küçük harfli yazabilir; sunucuya kanonik biçim gider.
///
/// `toUpperCase()` burada BİLEREK Türkçe kuralına çevrilmedi: kod alfabesi
/// yalnız A–Z ve 0–9'dur, süzgeç zaten Türkçe harfleri atar.
String kodTemizle(String? ham) =>
    (ham ?? '').toUpperCase().replaceAll(RegExp(r'[^A-Z0-9]'), '');

class PremiumCodeScreen extends StatefulWidget {
  const PremiumCodeScreen({super.key});

  @override
  State<PremiumCodeScreen> createState() => _PremiumCodeScreenState();
}

class _PremiumCodeScreenState extends State<PremiumCodeScreen> {
  Map<String, dynamic>? _durum;
  bool _yukleniyor = true;
  final _kod = TextEditingController();
  bool _gonderiliyor = false;
  ({String tur, String metin})? _mesaj; // tur: 'ok' | 'hata'

  @override
  void initState() {
    super.initState();
    _durumOku();
  }

  @override
  void dispose() {
    _kod.dispose();
    super.dispose();
  }

  Future<void> _durumOku() async {
    setState(() => _yukleniyor = true);
    try {
      final d = await api.premiumDurum();
      if (mounted) setState(() => _durum = (d as Map).cast<String, dynamic>());
    } catch (e) {
      if (mounted) {
        setState(() {
          _durum = null;
          _mesaj = (tur: 'hata', metin: '$e');
        });
      }
    } finally {
      if (mounted) setState(() => _yukleniyor = false);
    }
  }

  Future<void> _kullan() async {
    final temiz = kodTemizle(_kod.text);
    if (temiz.length < 4) {
      setState(() => _mesaj = (tur: 'hata', metin: 'Kodu eksiksiz yaz.'));
      return;
    }
    setState(() {
      _gonderiliyor = true;
      _mesaj = null;
    });
    try {
      final r = await api.premiumKodKullan(temiz);
      if (!mounted) return;
      setState(() {
        _durum = (r as Map).cast<String, dynamic>();
        _kod.clear();
        _mesaj = (
          tur: 'ok',
          metin: 'Kod kullanıldı — premium erişimin açıldı.',
        );
      });
    } catch (e) {
      if (mounted) setState(() => _mesaj = (tur: 'hata', metin: '$e'));
    } finally {
      if (mounted) setState(() => _gonderiliyor = false);
    }
  }

  /// Kaynakta `new Date(bitis).toLocaleDateString('tr-TR')` → "9.08.2026".
  String _tarih(Object? bitis) {
    if (bitis == null) return 'bilinmiyor';
    final d = DateTime.tryParse('$bitis')?.toLocal();
    if (d == null) return 'bilinmiyor';
    return '${d.day}.${d.month.toString().padLeft(2, '0')}.${d.year}';
  }

  @override
  Widget build(BuildContext context) {
    final premium = _durum?['premium'] == true;
    final kurulmadi = _durum?['kurulmadi'] == true;
    final suresiz = _durum?['suresiz'] == true;

    final aciklama = kurulmadi
        ? 'Premium sistemi bu sunucuda henüz kurulmadı.'
        : premium
        ? (suresiz
              ? 'Süresiz erişimin var.'
              : 'Bitiş: ${_tarih(_durum?['bitis'])}')
        : 'Elinde bir kod varsa aşağıya yazarak erişimini açabilirsin.';

    return Scaffold(
      backgroundColor: AppColors.bg,
      appBar: AppBar(title: const Text('Premium Kodu')),
      body: ListView(
        padding: const EdgeInsets.fromLTRB(
          Spacing.md,
          Spacing.md,
          Spacing.md,
          40,
        ),
        children: [
          Container(
            padding: const EdgeInsets.all(Spacing.md),
            margin: const EdgeInsets.only(bottom: Spacing.md),
            decoration: BoxDecoration(
              color: AppColors.card,
              borderRadius: BorderRadius.circular(AppRadius.md),
              border: Border.all(
                color: premium ? AppColors.warning : AppColors.border,
                width: premium ? 2 : 1,
              ),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  premium ? '⭐ Premium erişimin açık' : 'Premium erişimin yok',
                  style: const TextStyle(
                    color: AppColors.text,
                    fontSize: 15,
                    fontWeight: AppFont.black,
                  ),
                ),
                if (_yukleniyor)
                  const Padding(
                    padding: EdgeInsets.only(top: 8),
                    child: SizedBox(
                      width: 20,
                      height: 20,
                      child: CircularProgressIndicator(
                        strokeWidth: 2,
                        color: AppColors.primary,
                      ),
                    ),
                  )
                else
                  Padding(
                    padding: const EdgeInsets.only(top: 6),
                    child: Text(
                      aciklama,
                      style: const TextStyle(
                        color: AppColors.textSoft,
                        fontSize: 13,
                        height: 18 / 13,
                      ),
                    ),
                  ),
              ],
            ),
          ),
          Container(
            padding: const EdgeInsets.all(Spacing.md),
            margin: const EdgeInsets.only(bottom: Spacing.md),
            decoration: BoxDecoration(
              color: AppColors.card,
              borderRadius: BorderRadius.circular(AppRadius.md),
              border: Border.all(color: AppColors.border),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'Kodum var',
                  style: TextStyle(
                    color: AppColors.text,
                    fontSize: 15,
                    fontWeight: AppFont.black,
                  ),
                ),
                const SizedBox(height: 10),
                Semantics(
                  label: 'Premium kodu',
                  textField: true,
                  child: TextField(
                    controller: _kod,
                    textCapitalization: TextCapitalization.characters,
                    autocorrect: false,
                    maxLength: 24,
                    style: const TextStyle(
                      color: AppColors.text,
                      fontSize: 16,
                      fontWeight: AppFont.black,
                      letterSpacing: 2,
                    ),
                    decoration: InputDecoration(
                      counterText: '',
                      hintText: 'ÖRNEK: A7K2M9P4XR',
                      hintStyle: const TextStyle(color: AppColors.textMuted),
                      filled: true,
                      fillColor: AppColors.surfaceSoft,
                      isDense: true,
                      contentPadding: const EdgeInsets.symmetric(
                        horizontal: 12,
                        vertical: 11,
                      ),
                      border: OutlineInputBorder(
                        borderRadius: BorderRadius.circular(AppRadius.sm),
                        borderSide: const BorderSide(color: AppColors.border),
                      ),
                      enabledBorder: OutlineInputBorder(
                        borderRadius: BorderRadius.circular(AppRadius.sm),
                        borderSide: const BorderSide(color: AppColors.border),
                      ),
                      focusedBorder: OutlineInputBorder(
                        borderRadius: BorderRadius.circular(AppRadius.sm),
                        borderSide: const BorderSide(color: AppColors.border),
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 10),
                Opacity(
                  opacity: _gonderiliyor ? 0.6 : 1,
                  child: GestureDetector(
                    onTap: _gonderiliyor ? null : _kullan,
                    child: Container(
                      padding: const EdgeInsets.symmetric(vertical: 12),
                      alignment: Alignment.center,
                      decoration: BoxDecoration(
                        color: AppColors.primary,
                        borderRadius: BorderRadius.circular(AppRadius.sm),
                      ),
                      child: Text(
                        _gonderiliyor ? 'Kontrol ediliyor…' : 'Kodu kullan',
                        style: const TextStyle(
                          color: Color(0xFFFFFFFF),
                          fontSize: 14,
                          fontWeight: AppFont.black,
                        ),
                      ),
                    ),
                  ),
                ),
                if (_mesaj case final m?)
                  Padding(
                    padding: const EdgeInsets.only(top: 10),
                    child: Text(
                      m.metin,
                      style: TextStyle(
                        fontSize: 13,
                        fontWeight: AppFont.heavy,
                        color: m.tur == 'ok'
                            ? AppColors.success
                            : AppColors.accent,
                      ),
                    ),
                  ),
                const Padding(
                  padding: EdgeInsets.only(top: 10),
                  child: Text(
                    'Kodlar büyük/küçük harf ve tire farkına takılmaz; “a7k2-m9p4” ile “A7K2M9P4” aynıdır. '
                    'Bir kodu yalnız bir kez kullanabilirsin.',
                    style: TextStyle(
                      color: AppColors.textMuted,
                      fontSize: 11.5,
                      height: 16 / 11.5,
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
